src/table/utils/handle-key-press.js

```diff
@@ -127,33 +142,36 @@ export const bodyHandleKeyPress = (
   cell,
   selDispatch,
   isAnalysisMode,
-  setFocusedCellCoord
+  setFocusedCellCoord,
+  announce
 ) => {
   switch (evt.key) {
     case 'ArrowUp':
     case 'ArrowDown':
     case 'ArrowRight':
     case 'ArrowLeft': {
-      !isCtrlShift(evt) && moveFocus(evt, rootElement, cellCoord, selectionState, setFocusedCellCoord);
+      !isCtrlShift(evt) && moveFocus(evt, rootElement, cellCoord, selectionState, setFocusedCellCoord, announce);
       break;
     }
     // Space bar: Selects value.
     case ' ': {
       preventDefaultBehavior(evt);
-      cell?.isDim && isAnalysisMode && selectCell(selectionState, cell, selDispatch, evt);
+      cell?.isDim && isAnalysisMode && selectCell({ selectionState, cell, selDispatch, evt, announce });
       break;
     }
     // Enter: Confirms selections.
     case 'Enter': {
       preventDefaultBehavior(evt);
       isAnalysisMode && selectionState.api.confirm();
+      isAnalysisMode && announce({ keys: 'SNTable.SelectionLabel.SelectionsConfirmed' });
```

Comment:
Don't check this twice. And actually better to check `selectionState.api.isModal()`, since we don't want to confirm if we are not in selection mode

Reply:
✅ Fixed